nomad/mock/job.go

```diff
@@ -705,3 +705,38 @@ func BigBenchmarkJob() *structs.Job {
 
 	return job
 }
+
+// A multi-group, multi-task job with actions testing.
+func ActionsJob() *structs.Job {
+	job := MinJob()
+
+	for i := 0; i < 2; i++ {
+		tg := *job.TaskGroups[0]
```

Comment:
You're looking to make multiple task groups, so you have to `Copy` them or they end up sharing internal pointers.

```suggestion
		tg := job.TaskGroups[0].Copy()
```